CHECK: Compare IP addresses when checking for clones, not their realhost.

diff -r 319450a28444 ircd/m_check.c
--- a/ircd/m_check.c
+++ b/ircd/m_check.c
@@ -172,7 +172,7 @@
 
   for (lp = chptr->members; lp; lp = lp->next_member) {
     acptr = lp->user;
-    if (!strcmp(cli_user(cptr)->realhost, cli_user(acptr)->realhost)) {
+    if (!irc_in_addr_cmp(&cli_ip(cptr), &cli_ip(acptr))) {
       clones++;
     }
   }
@@ -181,7 +181,7 @@
   if (clones >= 2) {
     for (lp = chptr->members; lp; lp = lp->next_member) {
       acptr = lp->user;
-      if (!strcmp(cli_user(cptr)->realhost, cli_user(acptr)->realhost)) {
+      if (!irc_in_addr_cmp(&cli_ip(cptr), &cli_ip(acptr))) {
         cli_marker(acptr) = clones;
         count++;
         if (clones == count) {
